File: cpp/kumbangbadak.cpp
#pragma once
#include <bits/stdc++.h>
#include "hewan.cpp"
using namespace std;

class kumbangBadak : public hewan 
{
private:
    string spesies;
    int panjangTanduk;
public:
    kumbangBadak() {}
    kumbangBadak(string famili, string habitat, string jenisMakanan, string caraBerkembangbiak, int harapanHidup, makanan food, string spesies, int panjangTanduk)
        : hewan(famili, habitat, jenisMakanan, caraBerkembangbiak, harapanHidup, food) 
    {
        this->spesies = spesies;
        this->panjangTanduk = panjangTanduk;
    }

    // Setter
    void setSpesies(string s) { spesies = s; }
    void setPanjangTanduk(int p) { panjangTanduk = p; }

    // Getter
    string getSpesies() { return spesies; }
    int getPanjangTanduk() { return panjangTanduk; }

    void print() override 
    {
        hewan::print();
        cout << "Spesies            : " << getSpesies() << endl;
        cout << "Panjang Tanduk     : " << getPanjangTanduk() << " mm" << endl << endl;
    }
};

File: java/KumbangBadak.java
public class KumbangBadak extends Hewan {
    private String spesies;
    private int panjangTanduk;

    public KumbangBadak() {
        super();
    }

    public KumbangBadak(String famili, String habitat, String jenisMakanan, String caraBerkembangbiak, int harapanHidup, Makanan food, String spesies, int panjangTanduk) {
        super(famili, habitat, jenisMakanan, caraBerkembangbiak, harapanHidup, food);
        this.spesies = spesies;
        this.panjangTanduk = panjangTanduk;
    }

    // Setter
    public void setSpesies(String s) { spesies = s; }
    public void setPanjangTanduk(int p) { panjangTanduk = p; }

    // Getter
    public String getSpesies() { return spesies; }
    public int getPanjangTanduk() { return panjangTanduk; }

    @Override
    public void print() {
        super.print();
        System.out.println("Spesies            : " + getSpesies());
        System.out.println("Panjang Tanduk     : " + getPanjangTanduk() + " mm");
        System.out.println();
    }
}